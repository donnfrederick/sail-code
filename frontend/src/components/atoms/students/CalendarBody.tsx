import * as ConversationModels from 'models/conversation'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import getCalendarArray, { getDays } from 'utils/calendar'
import { isPast } from 'utils/isToday'
import zeroPadding from 'utils/zeroPadding'

interface Props {
  authToken: string
  calendar: ConversationModels.Calendar
  year: number
  month: number
  selectedDate: string
  getRecommend(authToken: string, date: string): void
  selectDate(date: string): void
}

export default (props: Props) => {
  const {
    authToken,
    calendar,
    year,
    month,
    selectedDate,
    getRecommend,
    selectDate
  } = props

  const weeks = getCalendarArray(year)(month)
  if (!weeks) {
    return <Container />
  }

  return (
    <Container>
      {getDays('en').map(day => (
        <DateContainer data-day-line="true" key={day}>
          {day}
        </DateContainer>
      ))}
      {weeks.map(week =>
        week.map(
          (date, index) =>
            date === 0 ? (
              <DateContainer key={year + month + index} />
            ) : (
              <DateContainer
                key={date}
                data-selected={
                  year === moment(selectedDate).year() &&
                  month === moment(selectedDate).month() + 1 &&
                  date === moment(selectedDate).date()
                }
                data-disabled={
                  isPast(year, month, date) ||
                  (calendar[date]
                    ? calendar[date].is_enabled
                      ? 'false'
                      : 'true'
                    : 'false')
                }
                data-status={
                  calendar[date]
                    ? calendar[date].is_reserved
                      ? 'reserved'
                      : null
                    : null
                }
                onClick={() => {
                  selectDate(
                    `${year}-${zeroPadding(month)}-${zeroPadding(date)}`
                  )
                  getRecommend(
                    authToken,
                    `${year}-${zeroPadding(month)}-${zeroPadding(date)}`
                  )
                }}
              >
                {date}
              </DateContainer>
            )
        )
      )}
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  flex-flow: wrap;
  justify-content: center;
  align-items: center;
  align-content: center;
  width: 100%;
  box-sizing: border-box;
  padding: 25px 0 40px;
`

const DateContainer = styled.div`
  position: relative;
  width: 62px;
  height: 62px;
  margin: 0 17px 24px;
  font-size: 32px;
  font-weight: 500;
  letter-spacing: 0.5px;
  line-height: 62px;
  text-align: center;
  color: #405766;

  &[data-day-line='true'] {
    font-size: 28px;
    font-weight: 500;
    color: #405766;
  }

  &[data-selected='true'] {
    width: 66px;
    height: 66px;
    margin: 0 15px 20px;
    border-radius: 50%;
    background-color: #138efd;
    line-height: 66px;
    color: #ffffff;
  }

  &[data-disabled='true'] {
    pointer-events: none;
    color: rgba(64, 87, 102, 0.3);
  }

  &[data-status='reserved'] {
    &::after {
      display: block;
      position: absolute;
      top: 62px;
      right: 0;
      left: 0;
      width: 12px;
      height: 12px;
      margin: auto;
      border-radius: 50%;
      background-color: #1cdfb9;
      content: '';
    }
  }
`
