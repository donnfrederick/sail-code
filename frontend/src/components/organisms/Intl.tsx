import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
