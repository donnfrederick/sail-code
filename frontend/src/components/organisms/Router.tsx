import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
