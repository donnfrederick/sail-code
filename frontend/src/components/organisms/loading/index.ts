import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
