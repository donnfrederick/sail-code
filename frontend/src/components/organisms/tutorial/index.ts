import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
