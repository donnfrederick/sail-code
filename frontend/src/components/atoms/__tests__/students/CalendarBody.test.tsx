import CalendarBody from 'components/atoms/students/CalendarBody'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <FooterButton />', () => {
  const tree = renderer
    .create(
      <Intl>
        <CalendarBody
          authToken={''}
          calendar={{
            '1': {
              is_enabled: false,
              is_reserved: false
            }
          }}
          year={1970}
          month={1}
          selectedDate={'2018-01-01'}
          getRecommend={() => null}
          selectDate={() => null}
        />
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
