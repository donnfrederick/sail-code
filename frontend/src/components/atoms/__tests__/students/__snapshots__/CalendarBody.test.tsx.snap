// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <FooterButton /> 1`] = `
<div
  className="sc-bdVaJa dqmDhL"
>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Sun
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Mon
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Tue
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Wed
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Thu
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Fri
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-day-line="true"
  >
    Sat
  </div>
  <div
    className="sc-bwzfXH gcJhme"
  />
  <div
    className="sc-bwzfXH gcJhme"
  />
  <div
    className="sc-bwzfXH gcJhme"
  />
  <div
    className="sc-bwzfXH gcJhme"
  />
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    8
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    13
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    14
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    15
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    16
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    17
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    18
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    19
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    21
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    22
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    23
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    24
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    25
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    26
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    27
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    28
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    29
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    30
  </div>
  <div
    className="sc-bwzfXH gcJhme"
    data-disabled={true}
    data-selected={false}
    data-status={null}
    onClick={[Function]}
  >
    31
  </div>
</div>
`;
